modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableRowSkinBase.java

```diff
@@ -349,12 +349,13 @@ protected ObjectProperty<Node> graphicProperty() {
                 height = h;
             }
 
-            width = tableCell.prefWidth(height);
-
             if (isVisible) {
                 if (fixedCellSizeEnabled && tableCell.getParent() == null) {
                     getChildren().add(tableCell);
                 }
+                // Note: We have to determine the pref width here because the add operation above may trigger the skin
+                // creation first, which is what makes it possible to get a correct value here in the first place.
+                width = tableCell.prefWidth(height);
```

Comment:
I wonder if it's safe to move this call (one one after else:430) before the if statement on line 352?

Reply:
I'm not sure if I understand correctly: You mean call `tableCell.prefWidth(height)` above?
I chose this way so we always call `tableCell.prefWidth(height)` just once, never twice.

Reply:
actually, please disregard my suggestion:  the first `width = tableCell.prefWidth(height);` happens after adding the `tableCell` to a parent, so it is likely would not be an equivalent change.